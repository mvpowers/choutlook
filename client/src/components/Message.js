import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Container = styled.div`
  padding: 10px 30px;
  border-bottom: 1px solid #eaeaea;
  font-size: 12px;
  word-wrap: break-word;

  &:hover {
    background-color: #eaeaea;
  }
`;

const Text = styled.div`
  color: #666;
`;

const Message = ({ user, msg }) => (
  <Container>
    <div>{user}@choutlook.com</div>
    <Text>{msg}</Text>
  </Container>
);

Message.propTypes = {
  user: PropTypes.string.isRequired,
  msg: PropTypes.string.isRequired,
};

export default Message;
