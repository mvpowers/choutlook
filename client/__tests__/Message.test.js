import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Message from '../src/components/Message';

Enzyme.configure({ adapter: new Adapter() });

describe('Message', () => {
  it('should be defined', () => {
    expect(Message).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = shallow(<Message user="user" msg="msg" />);
    expect(tree).toMatchSnapshot();
  });
});
