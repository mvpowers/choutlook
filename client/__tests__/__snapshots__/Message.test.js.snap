// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Message
    msg="msg"
    user="user"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div>
          user
          @choutlook.com
        </div>,
        <Message__Text>
          msg
        </Message__Text>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "user",
            "@choutlook.com",
          ],
        },
        "ref": null,
        "rendered": Array [
          "user",
          "@choutlook.com",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "msg",
        },
        "ref": null,
        "rendered": "msg",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div>
            user
            @choutlook.com
          </div>,
          <Message__Text>
            msg
          </Message__Text>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "user",
              "@choutlook.com",
            ],
          },
          "ref": null,
          "rendered": Array [
            "user",
            "@choutlook.com",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "msg",
          },
          "ref": null,
          "rendered": "msg",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
